src/components/CustomerAdd/CustomerAdd.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState } from 'react';
+import * as yup from 'yup';
+import { useHistory } from 'react-router-dom';
+
+import api from '../../api/api';
+
+export default function CustomerAd() {
+  const history = useHistory();
+
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [birthdate, setBirthDate] = useState('');
+  const [image, setImage] = useState('');
+  const [address, setAddress] = useState('');
+  const [city, setCity] = useState('');
+  const [state, setState] = useState('');
+  const [zip, setZip] = useState('');
+  const [error, setError] = useState('');
+
+  const validate = (data) => {
+    const schema = yup.object().shape({
+      first_name: yup.string().required(),
+      last_name: yup.string().required(),
+      birthdate: yup.date().required(),
+      address: yup.string(),
+      city: yup.string(),
+      zip: yup.string(),
+      state: yup.string(),
+      image: yup.string(),
```

Comment:
I'm good with what you have as required here, I can update our DB outline to allow NULL. This is the current specs:

Customers: will keep track of the customers that buy our products
id: int, auto increment, unique, not NULL, PK 
first_name: varchar(255), not NULL
last_name: varchar(255), not NULL
birthdate: date
address: varchar(255), not NULL
city: varchar(255), not NULL
state: varchar(255), not NULL
zip: varchar(255), not NULL
created_at: date, not NULL
updated_at: date, not NULL
image: varchar(255)

Reply:
I think `birthdate` can be optional too?

Reply:
Good with me @dlimx